API/Controllers/SearchController.cs

```diff
@@ -0,0 +1,58 @@
+using API.DTOs;
+using API.Services.Interfaces;
+using Asp.Versioning;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+[ApiController]
+[ApiVersion(1)]
+[EnableCors]
+[Authorize(Roles = "system")] // Internal access only at this time
+[AllowAnonymous]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class SearchController(ITournamentsService tournamentsService, IMatchesService matchesService, IPlayerService playerService) : Controller
+{
+    [HttpPost]
```

Comment:
```suggestion
    [HttpGet]
```
This should be a `GET` method